byway/src/main/java/com/google/sps/data/UserInfo.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/** A class to make a UserInfo type, containing an email, id and a list of trip IDs per user. */
+public final class UserInfo {
+
+  public static final String DATASTORE_ENTITY_KIND = "user";
+
+  private final String email;
+  private final String id;
+  private final ArrayList<Key> tripIds;
```

Comment:
Hm, storing Keys in datastore is probably better for querying the datastore, but worse for converting them with `gson`, since you don't really know how they'll end up being converted to gson.

If you want to use the same objects for gson serialization as for working with datastore entities, then it's probably OK to store them as strings.

Reply:
Updated as Strings. We won't have to query the datastore, since we will retrieve the stringified key from the front end and can access the same entity directly.